### PR TITLE
chore: upgrade packages to fix cves

### PR DESCRIPTION
**Pull request by yungbender**
CVE-2022-46908
CVE-2021-31239
CVE-2023-36191
CVE-2023-29401

upgraded sql migrate and gin

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: .github/workflows/run_linters.yaml [hunk 1]
```diff
@@ -21,4 +21,4 @@ jobs:
     - name: Lint
       uses: golangci/golangci-lint-action@v2
       with:
-        version: v1.45.1
+        version: v1.53.3
```
[the remark this answers is not in the document]
Comment (yungbender) [reply .github/workflows/run_linters.yaml]: Without the upgrade it returns

```
 Running [/home/runner/golangci-lint-1.45.1-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt
  
  goroutine 1 [running]:
  github.com/go-critic/go-critic/checkers.init.22()
  	github.com/go-critic/go-critic@v0.6.2/checkers/embedded_rules.go:46 +0x4b4
```

[the remark this answers is not in the document]
Comment (yungbender) [reply .github/workflows/run_linters.yaml]: Let's fix it by the upgrade of golangci-lint, check https://github.com/golangci/golangci-lint/issues/3107.

File: .vscode/launch.json
Comment (psegedy) [inline .vscode/launch.json:1]: do you want this file in the repo? no problem with it if that's intentional

File: digestwriter/storage_test.go [hunk 2]
```diff
@@ -234,8 +234,9 @@ func TestWriteClusterInfoWithExistingAccountForClusterName(t *testing.T) {
 
 // TestWriteClusterInfoNoAccountForClusterName tests that
 // Storage.WriteClusterInfo creates a new record in account table if
-//  no account can be retrieved for given clusterName and updates
-//  the cluster_image table properly
+//
+//	no account can be retrieved for given clusterName and updates
+//	the cluster_image table properly
```
Comment (psegedy) [inline digestwriter/storage_test.go:239]: this is looking weird 😄

**Review by psegedy — COMMENTED** [review]
check if you want `.vscode/launch.json` here, otherwise LGTM

File: .vscode/launch.json
Comment (yungbender) [reply .vscode/launch.json:1]: nope, that was my fault. I thought the `.vscode` is in `.gitignore` but it was not, so I added it there

File: digestwriter/storage_test.go (hunk 2 above)
Comment (yungbender) [reply digestwriter/storage_test.go]: fixed, some weird gofmt behaviour

**Pull request merged**